framework/tests/value/ValueCast.java

```diff
@@ -0,0 +1,11 @@
+// Test case for issue 1264: https://github.com/typetools/checker-framework/issues/1264
```

Comment:
To fix #1264, the Value Checker only needs to override org.checkerframework.common.basetype.BaseTypeVisitor#checkTypecastSafety and checkTypecastRedundancy so that cast unsafe warnings aren't issued.  

To fix this example, the declared type of `x` needs to be changed to `@IntRange(from=0, to = Short.MAX_VALUE)`.  I think the general problem in this example is that `@IntRange(from = 0) short` should be assumed to be `@IntRange(from = 0, to = Short.MAX_VALUE ) short`